File: Programmers/124 나라의 숫자.py
def solution(n):
    digit = {1: "1", 2: "2", 0: "4"}
    answer = ''
    while n > 0:
        mod = n % 3
        n = n // 3
        answer = digit[mod] + answer
        if mod == 0:
            n -= 1

    return answer

File: Programmers JS/124 나라의 숫자.js
function solution(n) {
  var answer = "";
  let match = {
    1: 1,
    2: 2,
    0: 4,
  };

  while (n !== 0) {
    let q = n % 3;
    answer = match[q].toString() + answer;
    n = parseInt(n / 3);
    if (q === 0) {
      n--;
    }
  }

  return answer;
}
